src/test/scala/io/delta/tables/DeltaTableSuite.scala

```diff
@@ -125,4 +126,21 @@ class DeltaTableSuite extends QueryTest
     val e = intercept[AnalysisException] { thunk }
     assert(e.getMessage.toLowerCase(Locale.ROOT).contains(expectedMsg.toLowerCase(Locale.ROOT)))
   }
+
+  test("DeltaTable is Java Serializable but cannot be used in executors") {
+    import testImplicits._
+
```

Comment:
could you also add a test that a closure refers to a DeltaTable object but doesn't use it, so that we can verify this does fix the issue? For example,
```
spark.range(5).as[Long].map{ row: Long =>
          dt
          row + 3
        }.show()
```